app/models/manageiq/providers/openstack/inventory/persister/definitions/cloud_collections.rb

```diff
@@ -150,6 +150,7 @@ def add_snapshots
     add_collection(cloud, :snapshots) do |builder|
       builder.add_properties(:model_class => ::Snapshot)
       builder.add_properties(:parent_inventory_collections => %i[vms miq_templates])
```

Comment:
@mansam so we might have a bad definition here. The parent_inventory collection says the snapshots are owned by VMs and images and we can't really do targeted refresh of those in MIQ. 

It's only possible to do a targeted refresh of Vm, which expects you will send all snapshots there, otherwise it will delete the snapshots you didn't send. (the query is fetching all snapshots of targeted VMs)

So the snapshot should really have ems_id and not be owned by a Vm or image (after all snapshot can live without VM or image), then you can ditch the parent_inventory_collections and targeted refresh should work as expected.

In the meantime, the `:complete => false` will prevent Vm targeted refresh to delete all snapshots that are not being sent for refresh, so I think it's a good workaround for now.

Reply:
@Ladas Ahh, gotcha. Thanks for the thorough explanation.